core/src/edu/cornell/gdiac/main/GameController.java

```diff
@@ -592,4 +755,206 @@ public long playSound(SoundEffect sound, long soundId, float volume) {
     }
     return sound.play(volume);
   }
+
+  // -----------------------------------------------------------------
+  // Game Screen UI Stages
+  // -----------------------------------------------------------------
+
+  /**
+   * Enum representing the state of this controller
+   * <p>
+   * Each represents a disjoint state that the controller can be in
+   */
+  public enum GameState {
+    INIT,
+    RUNNING,
+    PAUSED,
+    LEVEL_SUCCESS,
+    LEVEL_FAILURE
+  }
+
+  /**
+   * Game stage
+   */
+  private class GameMenu extends Stage {
+
+    boolean menuActive;
+
+    protected GameMenu(Viewport viewport, Batch batch) {
+      super(viewport, batch);
+
+      Table table = new Table();
+      table.setFillParent(true);
+
+      // TODO: Create overlay background
+
+      this.addActor(table);
+
+      menuActive = false;
+    }
+
+    public boolean isActive() {
+      return menuActive;
+    }
+
+    public void show() {
+      menuActive = true;
+      Gdx.input.setInputProcessor(this);
+    }
+
+    public void hide() {
+      menuActive = false;
+      // Reset to game controller input processor
+      Gdx.input.setInputProcessor(gameInputController.getInputProcessor());
+    }
+
+    @Override
+    public void act(float dt) {
+      if (menuActive && screenActive) {
+        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
+        super.act(dt);
+      }
+    }
+
+    @Override
+    public void draw() {
+      if (menuActive && screenActive) {
+        super.draw();
+      }
+    }
+
+    /**
+     * Game button factory method. Creates a new game button with specified style
+     */
+    protected GameButton createGameButton(String label, ClickListener clickListener) {
+      // Styling
+      ImageTextButtonStyle style = new ImageTextButtonStyle();
+      style.font = directory.getEntry("gameMenu", BitmapFont.class);
+
+      // Initialize and return new button
+      GameButton button = new GameButton(label, style);
+      button.addListener(clickListener);
+
+      return button;
+    }
+
+    /**
+     * Button containing optional icon image and label
+     */
+    private class GameButton extends ImageTextButton {
+
+      public GameButton(String label, ImageTextButtonStyle style) {
+        super(label, style);
+
+        // TODO: Add image in the future
+      }
+    }
+  }
+
+  /**
+   * Pause menu stage
+   */
+  private class PauseMenu extends GameMenu {
+
+    protected PauseMenu(Viewport viewport, Batch batch) {
+      super(viewport, batch);
+    }
+
+    protected void init() {
+      // Vertical layout
+      VerticalGroup verticalGroup = new VerticalGroup();
+      verticalGroup.space(10); // Set spacing between buttons
+      verticalGroup.setPosition(this.getWidth() / 2, this.getHeight() / 2);
+
+      this.addActor(verticalGroup);
```

Comment:
Thank you for doing the research on this and finding a good way to do this without hard coding any positions!